lib/Sema/CheckedCSubst.cpp

```diff
@@ -75,18 +76,254 @@ ExprResult Sema::ActOnTypeApplication(ExprResult TypeFunc, SourceLocation Loc,
 
 }
 
+// Type Instantiation
+RecordDecl* Sema::ActOnRecordTypeApplication(RecordDecl *Base, ArrayRef<TypeArgument> TypeArgs) {
+  assert(Base->isGeneric() && "Base decl must be generic in a type application");
+  assert(Base->getCanonicalDecl() == Base && "Base must be canonical decl");
+ 
+  auto &ctx = Base->getASTContext();
+
+  // Unwrap the type arguments from a 'TypeArgument' to the underlying 'Type *'.
+  auto RawArgs = new (ctx) llvm::SmallVector<const Type *, 4>();
```

Comment:
Is heap allocation of "RawArgs" really needed?

Reply:
I agree that it is likely not needed - good catch.

Reply:
Not needed. Allocated it on the stack.